dash/include/dash/GlobAsyncRef.h

```diff
@@ -315,203 +222,72 @@ class GlobAsyncRef
   /**
    * Asynchronously set the value referenced by this \c GlobAsyncRef
    * to the value pointed to by \c tptr.
-   * This operation is guaranteed to be complete after a call to \ref flush,
-   * but the pointer \c tptr can be re-used immediately.
+   * This operation is guaranteed to be complete after a call to \ref flush
+   * and the pointer \c tptr should not be reused before completion.
    */
-  void put(const_value_type* tptr) {
-    operator=(*tptr);
+  void set(const_value_type* tptr) {
+    DASH_LOG_TRACE_VAR("GlobAsyncRef.set()", *tptr);
+    DASH_LOG_TRACE_VAR("GlobAsyncRef.set()", _gptr);
+    dash::dart_storage<T> ds(1);
+    DASH_ASSERT_RETURNS(
+      dart_put(_gptr, static_cast<const void *>(tptr), ds.nelem, ds.dtype),
+      DART_OK
+    );
   }
 
   /**
    * Asynchronously set the value referenced by this \c GlobAsyncRef
-   * to the value pointed to by \c tref.
+   * to the value pointed to by \c new_value.
    * This operation is guaranteed to be complete after a call to \ref flush,
-   * but the value referenced by \c tref can be re-used immediately.
-   */
-  void put(const_value_type& tref) {
-    operator=(tref);
-  }
-
-  /**
-   * Value assignment operator, sets new value in local memory or calls
-   * non-blocking put on remote memory. This operator is only used for
-   * types which are comparable
-   */
-  template<typename __T = T>
-  typename std::enable_if<dash::has_operator_equal<__T>::value, self_t & >::type
-  operator=(const_value_type & new_value)
-  {
-    DASH_LOG_TRACE_VAR("GlobAsyncRef.=()", new_value);
-    DASH_LOG_TRACE_VAR("GlobAsyncRef.=", _gptr);
-    // TODO: Comparison with current value could be inconsistent
-    if (!_has_value || _value != new_value) {
-      _value       = new_value;
-      _has_value   = true;
-      if (_is_local) {
-        *_lptr = _value;
-      } else {
-        dash::dart_storage<T> ds(1);
-        DASH_ASSERT_RETURNS(
-          dart_put(
-            _gptr, static_cast<const void *>(&_value), ds.nelem, ds.dtype),
-          DART_OK
-        );
-      }
+   * but the value referenced by \c new_value can be re-used immediately.
+   */
+  void set(const_value_type& new_value) {
```

Comment:
Does this renaming break anything? Should we provide an alias `put()`?

Reply:
I will check. IMO, `put()` is closer to a C PGAS API.